Content.Client/UserInterface/Controls/RadialMenu.cs

```diff
@@ -242,14 +297,107 @@ private void OnClicked(ButtonEventArgs args)
         parent.TryToMoveToNewLayer(TargetLayer);
     }
 
+    /// <inheritdoc />
+    protected override bool HasPoint(Vector2 point)
+    {
+        var cX = -Position.X + Parent!.Width / 2;
+        var cY = -Position.Y + Parent.Width / 2;
+
+        var dist = Math.Sqrt(Math.Pow(point.X - cX, 2) + Math.Pow(point.Y - cY, 2));
+        var parent = (RadialContainer)Parent;
+        var isInRadius = dist < parent.Radius * 2 && dist > parent.Radius / 2;
+        if (!isInRadius)
+        {
+            return false;
+        }
+
+        var dX = cX - point.X;
+        var dY = cY - point.Y;
+        var angle = dX > 0
+            ? -MathF.Atan2(dX, dY) + MathF.PI * 2
+            : -MathF.Atan2(dX, dY);
```

Comment:
as i said in other comment - its all due to Y axis being flipped - i really dont want to dig it, i'm not used enough to working with graphics to efficiently cope with such problems >_> And it does not seem like its that critical - radial controls will improve MUCH more just from automating working of UI layer with model - just look at EmotesMenu c-tor, its bloated shit, 70% of code is stuff that should be solved by infrastructure and not high-level control itself...

Reply:
By checking in bad math, you're creating _two_ problems for the next person to work on this code, where now, they need to understand your code and then they need to fix it, before they can even get started on whatever they were trying to do.

I have made your math more conventional in https://github.com/eoineoineoin/space-station-14/commit/7ec22897a66c997cca42e433320e94429808cc15, if you want to cherry-pick that.

Reply:
There is an interesting side effect of this change set - it looks like sections rotated by a 30 degrees. I'm quite stunned how it was changed but it still works >_> need to look into math more...
![image](https://github.com/user-attachments/assets/6cac1929-2177-428f-a451-e9be7f414349)

Reply:
Okay its actually mirrored (that part i understand) and then rotated 90 degrees. I'm not sure how critical of a change that would be - people most likely got accustomed to current icon positions...

Reply:
I had to make some corrections - as chromiumboy said in discord - its better to have it oriented around Y-axis symmetry, not X-asix symmetry. So i added offset as button parameter. Checked - it works both ways, rotates stuff. yay.
And separator lanes logic was messed up with offset because of this offset. Oh well, now angle is negative here and all works well. I'd say that i kept most of your changes :3 so ty for contribution to our great cause.